src/config/mod.rs

```diff
@@ -0,0 +1,127 @@
+//! Provides a simple way of reading configuration files.
+//!
+//! Reads a TOML config file for the IPC Agent and deserializes it in a type-safe way into a
+//! [`Config`] struct.
+
+mod deserialize;
+mod server;
+mod subnet;
+
+use std::fs;
+
+use anyhow::Result;
+use serde::Deserialize;
+pub use server::Server;
+pub use subnet::Subnet;
+pub use server::JSON_RPC_ENDPOINT;
+pub use crate::config::subnet::Subnets;
+
+pub const JSON_RPC_VERSION: &str = "2.0";
+
+/// The top-level struct representing the config. Calls to [`Config::from_file`] deserialize into
+/// this struct.
+#[derive(Deserialize)]
+pub(crate) struct Config {
+    pub server: Server,
+    pub subnets: Subnets,
```

Comment:
nit: As mentioned in the other PR, I'd rather see here a `HashMap<String, Subnet>` instead of this custom type.

Reply:
my bad, I didn't see this. Will revert this to `HashMap`.